indra/assemblers/html/css_generator.py

```diff
@@ -0,0 +1,89 @@
+"""
+Assemble CSS rules to be inserted in an HTML file's style tag or to be saved
+in a separate CSS file.
+"""
+
+import logging
+from typing import Optional, Dict
+from os.path import dirname, abspath, join
+
+from jinja2 import Environment, FileSystemLoader, Template
+
+from .assembler import DEFAULT_SOURCE_COLORS
+
+logger = logging.getLogger(__name__)
+HERE = dirname(abspath(__file__))
+
+
+# Note that if templates of the same name exist in both 'templates/indra' and
+# 'templates/macros', the template in the first directory will be used.
+loader = FileSystemLoader([join(HERE, 'templates', 'indra'),
+                           join(HERE, 'templates', 'macros')])
+env = Environment(loader=loader)
+
+stylesheet_template_file = 'standalone_stylesheet_template.css'
+
+
+class CSSGenerator:
+    """Base class to create a stylesheet from a jinja2 template"""
+    template_file = stylesheet_template_file
+
```

Comment:
You can annotate the instance variables inside the class body - this lets other code introspect by asking for `CSSGenerator.__annotations__`:

```suggestion

template: Template
model: Optional[str]
template_kwargs: Dict

```

Reply:
Then you don't have to do the annotations inside the executed code